cloudify_rest_client/secrets_providers.py

```diff
@@ -100,6 +105,46 @@ def create(
 
         return SecretsProvider(response)
 
+    def update(
+            self,
+            name,
+            _type,
+            connection_parameters,
+            visibility=VisibilityState.TENANT,
+    ):
+        """
+        Update already exists Secrets Provider.
+
+        :param name: Secrets Provider name
+        :type name: str
+        :param _type: Secrets Provider type
+        :type _type: str
+        :param connection_parameters: Secrets Provider connection parameters
+        :type connection_parameters: dict
+        :param visibility: The visibility of the secret, can be 'private',
+                           'tenant' or 'global'
+        :type visibility: str
+        :returns: New secrets provider metadata
+        :rtype: Dict[str]
```

Comment:
`Dict[str]` isn't a type, `Dict[str, Any]` would be a type, but really, I think we can just say...
```suggestion
        :rtype: SecretsProvider
```